src/dotnet/commands/dotnet-restore3/Program.cs

```diff
@@ -97,6 +103,13 @@ public static int Run(string[] args)
                     msbuildArgs.Add($"/p:RestoreIgnoreFailedSources=true");
                 }
 
+                // By default restore is recursive, if set to false turn it off.
```

Comment:
Sure, you can merge this as is.  I'll just rename the option after you merge.

The CLI as a whole is an API.  It shouldn't appear to customers that 5 different teams are contributing to it, each with their own naming preferences, etc.  If we have an option in `build` and the exact same option in `restore`, they should be named the same. That way our customers don't get confused.  API consistency is an important feature.

/cc @blackdwarf